client/src/components/ImageHandler.tsx

```diff
@@ -0,0 +1,98 @@
+import 'react-image-crop/dist/ReactCrop.css';
+
+import { useRef, useState } from 'react';
+import { twMerge } from 'tailwind-merge';
+
+import { ImageEditDialog } from './dialogs/ImageEditDialog';
+import { Button } from './ui/button';
+
+interface ImageHandlerProps {
+  callback?: (imageBlob: Blob, imageURL: string) => void;
+  removeSelectionCallback: () => void;
+  width: number;
+  height: number;
+  handlerId: string;
+  className?: string;
+}
+
+// callback will be run after the image is saved and is used to synchronize the component with its environment
+export function ImageHandler({
+  callback,
```

Comment:
Let's call this `onImageSave` maybe? I think the name would be more meaningful and comment wouldn't be necessary